porting/debug-build/faq.rst

```diff
@@ -0,0 +1,131 @@
+
+Porting FAQ
+===========
+
+This is a list of common error messages and corresponding fixes people run into when porting new devices to Halium. 
+Contributions welcome!
+
+Getting ``kgsl kgsl-3d0: |_load_firmware| request_firmware(a330_pm4.fw) failed: -2 … kgsl kgsl-3d0: |adreno_init| Reading pm4 microcode failed a330_pm4.fw``
+------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+Toggle kernel option ``CONFIG_FW_***LOADER``
+
+Dmesg shows systemd errors when running ``switch_root /target /sbin/init``
+--------------------------------------------------------------------------
+
+.. code-block:: guess
+
+   systemd[1]: Failed to determine whether /sys/fs/cgroup is a mount point: Too many levels of symbolic links
+   systemd[1]: Failed to mount cgroup at /sys/fs/cgroup/systemd: NO such filE oR directory
+   systemd[1]: Freezing execution.
+
+There's a patch on hammerhead kernel from Linus Torvalds you need to apply: [PATCH] vfs: Fix /proc/<tid>/fdinfo/<fd> file handling
+https://github.com/Halium/android_kernel_lge_hammerhead/commit/25437b2a54dd619a96e268ecaf303b089aa785e4
+
+
+Compiler error for kvfree with apparmor directory
+-------------------------------------------------
+
+.. code-block:: guess
+
+   ../../../../../../kernel/samsung/s3ve3g/security/apparmor/apparmorfs.c: In function 'aa_simple_write_to_buffer': 
+   ../../../../../../kernel/samsung/s3ve3g/security/apparmor/apparmorfs.c:110:3: error: implicit declaration of function 'kvfree' [-Werror=implicit-function-declaration]
+   kvfree(data);
+
+Use this patch:
+https://github.com/ubports/android_kernel_moto_shamu/commit/83f949a8de673fe45499d1741da8654831a5afae
+
+Build errors for systemimage which is trying to build an apk
+------------------------------------------------------------
+
+Remove all apks from the makefiles in ``vendor/$vendor/$device``
+Also check the vendor folders of your dependencies.
+
+My device is constantly rebooting. How do I get a new last_kmsg?
+----------------------------------------------------------------
+
+Enable the following configs:
+
+.. code-block:: guess
```

Comment:
If it's unlikely that Pygments has a lexer for a language (which is probably the case for `dmesg`, kernel configs, and Upstart config snippets), you can drop the `guess` attribute.

If there is a paragraph before a code block, end it with two colons and drop the `.. code-block::` directive. For example,

```
Enable the following configs::

    CONFIG_ANDROID_RAM_CONSOLE=y
```

That will have the same effect.